Gems/Multiplayer/Code/Source/MultiplayerStatSystemComponent.cpp

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Source/MultiplayerStatSystemComponent.h>
+
+namespace Multiplayer
+{
+    // Metrics cvars
+    void OnEnableNetworkingMetricsChanged(const bool& enabled);
+    AZ_CVAR(
+        bool,
+        bg_enableNetworkingMetrics,
+        true,
+        &OnEnableNetworkingMetricsChanged,
+        AZ::ConsoleFunctorFlags::DontReplicate,
+        "Whether to capture networking metrics");
+    AZ_CVAR(
+        AZ::TimeMs,
+        bg_networkingMetricCollectionPeriod,
+        AZ::TimeMs{ 1000 },
+        nullptr,
+        AZ::ConsoleFunctorFlags::DontReplicate,
+        "How often to capture metrics by default.");
+    AZ_CVAR(
+        AZ::CVarFixedString,
+        cl_metricsFile,
+        "client_network_metrics.json",
+        nullptr,
+        AZ::ConsoleFunctorFlags::DontReplicate,
+        "File of the client metrics file if enabled, placed under <ProjectFolder>/user/metrics");
+    AZ_CVAR(
+        AZ::CVarFixedString,
+        sv_metricsFile,
+        "server_network_metrics.json",
+        nullptr,
+        AZ::ConsoleFunctorFlags::DontReplicate,
+        "File of the server metrics file if enabled, placed under <ProjectFolder>/user/metrics");
+
+    void ConfigureEventLoggerHelper(const AZ::CVarFixedString& filename)
+    {
+        if (auto eventLoggerFactory = AZ::Interface<AZ::Metrics::IEventLoggerFactory>::Get())
+        {
+            const AZ::IO::FixedMaxPath metricsFilepath = AZ::IO::FixedMaxPath(AZ::Utils::GetProjectPath()) / "user/Metrics" / filename;
```

Comment:
Should these metrics filenames add a timestamp to their name or are you looking into going with a rotating log system?

Reply:
I'll work on adding later if it is needed by AWSI.